pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -76,14 +76,107 @@ pub(crate) enum VectoredReadExtended {
     No,
 }
 
-pub(crate) struct VectoredReadBuilder {
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum VectoredReadCoalesceMode {
+    /// Only coalesce exactly adjacent reads.
+    AdjacentOnly,
+    /// In addition to adjacent reads, also consider reads whose corresponding
+    /// `end` and `start` offsets reside at the same chunk.
+    Chunked(usize),
+}
+
+impl VectoredReadCoalesceMode {
+    fn get() -> Self {
+        let align = virtual_file::get_io_buffer_alignment();
+        if align == 1 {
+            VectoredReadCoalesceMode::AdjacentOnly
+        } else {
+            VectoredReadCoalesceMode::Chunked(align)
+        }
+    }
+}
+
+pub(crate) enum VectoredReadBuilder {
+    Unconstrained(VectoredReadBuilderInner),
```

Comment:
nit: not sure what `Unconstrained` means here. It can be constrained with `max_read_size`.